src/clients/legacy-service.js

```diff
@@ -94,4 +94,26 @@ export class LegacyServiceClient {
 
     return body.data;
   }
+
+  async getPublishingProperties(appId) {
+    const settings = {
+      json: true,
+      resolveWithFullResponse: true,
+      method: 'GET',
+      uri: new URI(this.serviceUri).segment(`/api/applications/publishing_properties.json`).search({ nid: appId }).toString(),
+      headers: {
+        Accept: 'application/json',
+        Authorization: `Bearer ${this.apiToken}`,
+        'Content-Type': 'application/json',
+      }
+    };
+
+    const res = await request2(settings);
```

Comment:
request2 doesn't say much about that variable. Rename it to requestPromise.
Even better solution would be to remove request and use request-promise everywhere to be consistent.